internal/dcache/rpc/server/handler.go

```diff
@@ -3456,6 +3467,250 @@ func (h *ChunkServiceHandler) GetMVSize(ctx context.Context, req *models.GetMVSi
 	return resp, nil
 }
 
+// GetLogs streams the tar.gz of blobfuse2.log* files in 16MB chunks.
+func (h *ChunkServiceHandler) GetLogs(ctx context.Context, req *models.GetLogsRequest) (*models.GetLogsResponse, error) {
+	// Thrift should not be calling us with nil req.
+	common.Assert(req != nil)
+
+	if common.IsDebugBuild() {
+		start := time.Now()
+		defer func() {
+			log.Debug("ChunkServiceHandler::GetLogs: request took %v: %v",
+				time.Since(start), req.String())
+		}()
+	}
+
+	log.Debug("ChunkServiceHandler::GetLogs: Received GetLogs request: %v", req.String())
+
+	if !common.IsValidUUID(req.SenderNodeID) ||
+		req.ChunkIndex < 0 ||
+		req.NumLogs <= 0 ||
+		req.ChunkSize <= 0 ||
+		req.ChunkSize > rpc.MaxLogChunkSize {
+		errStr := fmt.Sprintf("Invalid GetLogs request: %v", req.String())
+		log.Err("ChunkServiceHandler::GetLogs: %s", errStr)
+		common.Assert(false, errStr)
+		return nil, rpc.NewResponseError(models.ErrorCode_InvalidRequest, errStr)
+	}
+
+	//
+	// Take logStateMu lock to prevent multiple threads from creating the log tarball.
+	//
+	h.logStateMu.Lock()
+
+	//
+	// Create the tarball on first chunk or,
+	// if the logTarPath is empty (client asked for non-zero chunk index without initiating)
+	//
+	if req.ChunkIndex == 0 || h.logTarPath == "" {
```

Comment:
don't need to support clients starting at non-zero index.
it's our clien and we control that